### ISSUE TITLE
Understanding lifecycle methods in React.js

### ISSUE THREAD

**Issue by Neema-2016** [section-engineering-education/engineering-education#3624]
We ask that you please be patient as our team works through approving and publishing all articles/tutorials in a timely manner. Allow 1-3 days for a topic to be reviewed and/or approved - allow 3-7 days for an articles to be reviewed and/or published.

Be sure to visit our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for tools, resources, and example articles to go over.

**NOTE: (Please ensure that you have only one open issue + linked pull request at a time. This will ensure that we complete the article in a timely manner from inception to publishing.)**

**NOTE: We tend to stray away or tend not to publish reviews/comparisons of commercial product offerings.**

*We're looking for the first 2-3 paragraphs of the article that appropriately summarize what your article will be about.*

### Proposed title of article
Understanding lifecycle methods in React.js

### Introduction paragraph (2-3 paragraphs):
React.js is a front-end framework used to create user interfaces using components, that break down the complex UI of the application into simple reusable components. React.js uses either class-based or functional components. 
The lifecycle methods are implemented mainly in class-based components. These methods tend to implement the different phases a class component goes through from its initialization or in layman's; 'birth' which happens the first time it is rendered on the browser, to when this component is destroyed or unmounted. 
These methods include ;
1. componentWillMount()
2. componentWillUpdate()
3. componentWillUnmount() 
This article will bring the user to a better and practical understanding of these methods with hands-on code.

### Key takeaways: 
1. What lifecycle methods are and where they are used.
2. Phases of lifecycle methods.
3. Implementation of the lifecycle methods

### References: 
N/A

### Templates to use as guides
- [How To Guide Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/how-to-guide-template.md)
- [Software Review Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/software-review-template.md)
- [Tutorial Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/tutorial-template.md)

**Comment by WanjaMIKE** [comment]
Good afternoon and thank you for submitting your topic suggestion. @Neema-2016 
Your topic form has been entered into our queue and should be reviewed (for approval) as soon as a content moderator is finished reviewing the ones in the queue before it.

**Comment by Neema-2016** [comment]
Realized the issue is similar to #3529 .
